### PR TITLE
Bump transaction version default to 2

### PR DESCRIPTION
**Pull request by btcdrak**
This should be merged after the BIP68 softfork enforces.

EDIT: BIP68 has long since activated so miners are running min 0.12.1 and therefore will mine v2 txs. Also there are sufficient nodes on the network relaying v2 txs

### CODE REVIEW

**Comment by paveljanik** [comment]
After bumping the default tx version to 2, we should also have some test to test v1 transactions.

**Comment by laanwj** [comment]
Needs rebase (but probably only after it is eligible for merge in the first place)

**Comment by arowser** [comment]
Can one of the admins verify this patch?

**Comment by laanwj** [comment]
@btcdrak Is there anything still blocking this? (apart from needing rebase)

**Comment by MarcoFalke** [comment]
Does this need mention in the release notes?

File: src/test/txvalidationcache_tests.cpp [hunk 1]
```diff
@@ -39,6 +39,7 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
     spends.resize(2);
     for (int i = 0; i < 2; i++)
     {
+        spends[i].nVersion = 1;
```
Comment (btcdrak) [inline src/test/txvalidationcache_tests.cpp]: These tests break when using txversion=2 and I don't see why.

**Comment by paveljanik** [comment]
What is the status of this?

Needs rebased.

**Comment by laanwj** [comment]
@sipa @gmaxwell any opinion here? Can this be merged for 0.14?

**Comment by gmaxwell** [comment]
utACK. I think this is fine now!

**Comment by morcos** [comment]
ugh, thats a lot of work.
thanks @btcdrak 

I reviewed everything so far except the changes in test/data

technically the mempool_tests change was a bug in the test, here's a fix b095828 if you want it that means you don't need to add the version.

**Comment by morcos** [comment]
OK I've gone through test/data for as long as I can stand it.
utACK

**Comment by btcdrak** [comment]
> ugh, thats a lot of work.

@morcos was a lot more than I bargained for, to change a 1 to a 2 :laughing: 

I've added your commit thanks for that as it simplifies things a bit. Squashed.

**Comment by fanquake** [comment]
Travis failure:
```
2016-12-07 22:53:39,532 - ERROR - Output data mismatch for txcreate2.hex (format hex)
2016-12-07 22:53:44,125 - ERROR - Output data mismatch for txcreate2.json (format json)
2016-12-07 22:54:44,785 - ERROR - FAILED TESTCASES: [Parses a transation with no inputs and a single output script, Parses a transation with no inputs and a single output script (output in json)]
```

**Comment by btcdrak** [comment]
@fanquake I'm stuck, I cant replicate the fails locally.

**Comment by fanquake** [comment]
I'm able to recreate locally
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS check-local
Running test/bitcoin-util-test.py...
2016-12-08 18:40:39,202 - ERROR - Output data mismatch for txcreate2.hex (format hex)
2016-12-08 18:40:39,207 - ERROR - Output data mismatch for txcreate2.json (format json)
2016-12-08 18:40:39,307 - ERROR - FAILED TESTCASES: [Parses a transation with no inputs and a single output script, Parses a transation with no inputs and a single output script (output in json)]
make[3]: *** [check-local] Error 1
make[3]: *** Waiting for unfinished jobs....
```

**Comment by paveljanik** [comment]
I'm not able to replicate locally.

Sorry, couldn't resist: some typo collector should grab "transation".

**Comment by btcdrak** [comment]
I reverted the addition of b095828 because Travis was playing up. We can do that in a separate PR.

**Comment by btcdrak** [comment]
OK after investigation from @morcos and others found something had changed in master, so I had to rebase to master and fix a couple of new tests that had been introduced. Travis should pass now, and I've again included @morcos patch.

**Comment by droark** [comment]
Hello. Going through this PR, I noticed that a couple of files now seem to not be used. Is this intentional? I think a couple of them could be reused but a couple seem like they could be deleted.

- src/test/data/blanktx.json  (Doesn't seem to be used, although it could presumably be given a v1 rename and used for a v1 test.)
- src/test/data/txcreatesign.json  (Exact copy of txcreatesignv1.json, so it's redundant.)

I'm happy to file a PR and fix this. Just want to make sure I'm on the right path first. :)

Thanks.

**Comment by btcdrak** [comment]
@droark +1

**Comment by btcdrak** [comment]
@droark I had to knock out a quick PR because the orphan reference was causing someone an issue on IRC, see #9376

**Pull request merged**